pkg/cli/codearea.go

```diff
@@ -196,6 +200,71 @@ func (w *codeArea) handlePasteSetting(start bool) bool {
 	return true
 }
 
+// Try to expand a simple abbreviation that ignores word boundaries.
+func (w *codeArea) expandSimpleAbbr() {
+	var abbr, full string
+	// Find the longest matching abbreviation.
+	w.Abbreviations(func(a, f string) {
+		if strings.HasSuffix(w.inserts, a) && len(a) > len(abbr) {
+			abbr, full = a, f
+		}
+	})
+	if len(abbr) > 0 {
+		c := &w.State.Buffer
+		*c = CodeBuffer{
+			Content: c.Content[:c.Dot-len(abbr)] + full + c.Content[c.Dot:],
+			Dot:     c.Dot - len(abbr) + len(full),
+		}
+		w.resetInserts()
+	}
+}
+
+// Try to expand a small word abbreviation.
+func (w *codeArea) expandSmallWordAbbr(triggerRune rune, triggerLen int) {
```

Comment:
There is no need to pass `triggerLen` as it can be entirely determined by `triggerRune`.

Furthermore, this function can take the categorizer as an argument to make it work for other types of word boundaries in future.

Reply:
I included the rune's encoded length because we had already converted the rune to a UTF-8 sequence and it seemed silly to do it a second time in this function. I didn't benchmark the cost so this could very well be an example of an unneeded optimization.